src/main/java/io/jenkins/plugins/appcenter/AppCenterRecorder.java

```diff
@@ -100,11 +101,26 @@ private boolean uploadToAppCenter(Run<?, ?> run, FilePath filePath, TaskListener
         final PrintStream logger = taskListener.getLogger();
 
         try {
-            final AppCenterServiceFactory appCenterServiceFactory = new AppCenterServiceFactory(
-                    getApiToken(), getOwnerName(), getAppName(), getPathToApp(), getBaseUrl()
-            );
-
-            return filePath.act(new UploadTask(filePath, taskListener, appCenterServiceFactory));
+        	EnvVars vars = run.getEnvironment(taskListener);
+        	String pathParsed = vars.expand(getPathToApp());
+        	logger.println("Path is " + pathParsed);
+        	FilePath localFiles[] = filePath.list(pathParsed);
+        	if(localFiles.length == 0) {
+        		logger.println("No file found to upload: " + pathParsed);
+        		return false;
+        	}
+        	boolean result = true;
+        	ArrayIterator localFilesIterator = new ArrayIterator(localFiles);
```

Comment:
I don't want to support multiple uploads in this way..... or at all to be honest. If you need to upload multiple files I recommend calling the plugin multiple times instead. In the future we need to upload obfuscation files along with the app itself. Managing a collection of potentially differing sizes isn't something I want to do.